iox_catalog/migrations/20220607102200_change_sort_key_type_to_array.sql

```diff
@@ -0,0 +1,2 @@
+ALTER TABLE partition DROP COLUMN sort_key;
+ALTER TABLE partition ADD COLUMN sort_key TEXT[];
```

Comment:
Once #4826 gets merged, we'll have a `sort_key_arr` column. That column must be first manually migrated from `sort_key`. We'll leave the old `sort_key` column around so we can easily backfill the new sort_key even if this PR gets merged before we manually migrated the old `sort_key` into `sort_key_arr`.

```suggestion
ALTER TABLE partition RENAME COLUMN sort_key TO sort_key_old;
ALTER TABLE partition RENAME COLUMN sort_key_arr TO sort_key;
ALTER TABLE partition ALTER COLUMN sort_key SET NOT NULL;
```